### PR TITLE
Ibrahioy code review

### PR DESCRIPTION
**Pull request by oibrahim3**
Technical Changes:
1. Added Logger to try catch blocks with messages and fixed typos
2.  Fixed the created html page name. Added index because I saw a method that returned it.
3. Added NavBar to fragments and included it in both html pages created
4. Changed RequestParmater to method types. Example: (Put/Post/Get)Parmater
5. Renamed a variable in GameDTO to a more proper name along with its getter/setter and constructor

Analysis:
I couldn't compile and run the repository when i cloned it to eclipse which made it harder to see if everything was working fine and it prompted me for an error for autocomplete a couple of times. Overall great job from a coding perspective and some good progress made in my opinion. I liked that you had firewall authentication in there and I did learn some things from it as well as "Optional". Additionally, layers are separated properly and well documented and easy to read. The repository was available on GitHub in time. Again, I think my changes should compile but I was unable to run the program.

### CODE REVIEW

**Comment by vuhi** [comment]
The config file needs to be set up in order for the app to compile. We did say it in the readme.

**Comment by oibrahim3** [comment]
I did that and I don't think the problem is there. It's more of an eclipse not knowing what to do with it based on the teachers last announcement answer to question 4 of the QQS. When I cloned the project there was no option to run as a java application or anything at all.

**Comment by oibrahim3** [comment]
Also, I don't think it should effect my code review. I'm open to questions and feedback on any of my commits.

**Comment by Slashroot101** [comment]
It definitely compiles. If you can provide the error output that you are getting that would help with trying to figure it out. The app will not startup/build without the config files because part of the startup process is grabbing the config file for dependency injection

**Comment by Slashroot101** [comment]
Will likely merge this PR, but need to review the changes closer before we decide to.

File: src/main/java/edu/uc/gamelibrarymanager/GameController.java [hunk 1]
```diff
@@ -10,41 +10,48 @@
...
-    @RequestMapping("/index")
+    @GetMapping("/index")
     public String index() {
         return "index";
     }
 
     @PostMapping("/")
     ResponseEntity createGame(@RequestBody @Valid GameDTO newGame){
         try {
-            Assert.notNull(gameService, "Game service is null -- check DI container");
```
Comment (discospiff) [inline src/main/java/edu/uc/gamelibrarymanager/GameController.java]: DI is likely correct here: Dependency Injection.

**Comment by discospiff** [comment]
Several good suggestions here, but I would merge them manually.  Since these changes were created on a branch that does not compile, there's a chance that the changes do not compile as well... which is why they should be reviewed and implemented manually.

**Comment by Slashroot101** [comment]
I am going to manually merge this one at a later time. Leaving it open so I do not forget about it. @vuhi  Can you implement some of the HTML changes he made, if you feel they are appropriate?

**Pull request closed without merging**